src/handlers/swap_handlers.ts

```diff
@@ -63,6 +64,8 @@ export class SwapHandlers {
                 gasPrice,
                 excludedSources,
                 affiliateAddress,
+                apiKey: takerAddress && takerAddress !== NULL_ADDRESS ? req.header('0x-api-key') : undefined,
```

Comment:
Cut the business logic and just pass through the `apiKey`.  Enforce that requirement elsewhere (if it's not already addressed elsewhere).

Reply:
This enforcement is already in place in https://github.com/0xProject/0x-monorepo/pull/2541 in `SwapQuoter._getSwapQuoteAsync()`.

Reply:
Addressed in https://github.com/0xProject/0x-api/pull/162/commits/b1263163c60edcbc81cf5c5ac26f6c8b439e09bf